src/main/scala/com/atomist/tree/pathexpression/ExecutionResult.scala

```diff
@@ -0,0 +1,50 @@
+package com.atomist.tree.pathexpression
+
+import com.atomist.graph.{AddressableGraphNode, GraphNode}
+
+/**
+  * Result type and utility methods of a node navigation.
+  */
+object ExecutionResult {
```

Comment:
I agree that `EvaluationResult` might be better, but this will break the runner as it's a public API. Let's consider after this PR if we want to do that now.